src/Unosquare.RaspberryIO.Peripherals/AccelerometerGY521.cs

```diff
@@ -0,0 +1,205 @@
+
+    /// <summary>
+    /// Enum FSSEL
+    /// </summary>
+    internal enum FSSEL
+    {
+        /// <summary>
+        /// Gyroscope Full Scale Range 250
+        /// </summary>
+        FSR250,
+
+        /// <summary>
+        /// Gyroscope Full Scale Range 500
+        /// </summary>
+        FSR500,
+
+        /// <summary>
+        /// Gyroscope Full Scale Range 1000
+        /// </summary>
+        FSR1000,
+
+        /// <summary>
+        /// Gyroscope Full Scale Range 2000
+        /// </summary>
+        FSR2000,
+    }
+
+    /// <summary>
+    /// Enum AFSSEL
+    /// </summary>
+    internal enum AFSSEL
+    {
+        /// <summary>
+        /// Accelerometer Full Scale Range 2g
+        /// </summary>
+        FSR2G,
+
+        /// <summary>
+        /// Accelerometer Full Scale Range 4g
+        /// </summary>
+        FSR4G,
+
+        /// <summary>
+        /// Accelerometer Full Scale Range 8g
+        /// </summary>
+        FSR8G,
+
+        /// <summary>
+        /// Accelerometer Full Scale Range 16g
+        /// </summary>
+        FSR16G,
+    }
+
+    /// <summary>
+    /// Implements settings for GY-521 accelerometer peripheral.
+    /// </summary>
+    public sealed class AccelerometerGY521 : IDisposable
+    {
+        private const int PwrMgmt1 = 0x6b;
+        private const int GyroSensitivity = 0x1b;
+        private const int AcclSensitivity = 0x1c;
+
+        private readonly ushort[] AFSSel = new ushort[] { 0x4000, 0x2000, 0x1000, 0x0800 };
+        private readonly byte[] FSSel = new byte[] { 0x83, 0x42, 0x20, 0x10 };
+
+        private readonly Thread ReadWorker;
+        private readonly TimeSpan ReadTime = TimeSpan.FromSeconds(2);
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AccelerometerGY521"/> class.
+        /// </summary>
+        /// <param name="device"> i2C device. </param>
+        public AccelerometerGY521(II2CDevice device)
+        {
+            Device = device;
+
+            // Reset sensor sleep mode to 0.
+            Device.WriteAddressByte(PwrMgmt1, 0);
+            Device.WriteAddressByte(GyroSensitivity, FSSel[(int)FSSEL.FSR250]);
+            Device.WriteAddressWord(AcclSensitivity, AFSSel[(int)AFSSEL.FSR2G]);
+
+            ReadWorker = new Thread(Run);
+        }
+
+        /// <summary>
+        /// Occurs when [data available].
+        /// </summary>
+        public event EventHandler<AccelerometerGY521EventArgs> DataAvailable;
+
+        /// <summary>
+        /// Retrieves the I2C Device.
+        /// </summary>
+        public II2CDevice Device { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether this instance is running.
+        /// </summary>
+        /// <value><c>true</c> if this instance is running; otherwise, <c>false</c>.</value>
+        public bool IsRunning { get; private set; }
+
+        /// <summary>
+        /// Sets the sleep mode.
+        /// </summary>
+        public void SetSleepMode()
```

Comment:
This only will be called when the object is disposed, so must be a private function.